### ISSUE TITLE
Problems on comparing binary types (bit from MySQL)

### ISSUE THREAD

**Issue by NathanPB** [JetBrains/Exposed#491]
Hello everyone, I'm not sure if this is a bug or just me being dumb again.
Well, I have the following table:
```
create table Permissions (
  guild varchar(32),
  role varchar(32),
  permission varchar(128) not null,
  type bit(1) not null default 0,

  foreign key(guild) references Guilds(id)
);
```
... and I need to select the entries where ``type is 1``.
I'm trying
```
transaction {
                    return@transaction Database.PERMISSIONS.run {
                        slice(permission)
                        .select {
                            ((guild eq this@PermGroup.guild?.id) and
                            (role eq this@PermGroup.role?.id)) and
                            (type eq byteArrayOf(1)).also { println(it) }
                        }.map { it[permission] }
                    }
                }
```
but I got no results, then I added the ``.also { println(it) }`` statement to see the query and I got the following:
``permissions.guild IS NULL and permissions.role IS NULL and permissions.type = [B@2e877764``.

I think the ByteArray just isn't being serialized, and I honestly don't know how to deal with it.

Kotlin 1.3.21
Exposed 0.12.1
MySQL Connector 5.1.6
MySQL 5.7.23

**Comment by Tapac** [comment]
Could you specify how do you map `type`  column in Exposed?

**Comment by NathanPB** [comment]
> Could you specify how do you map `type` column in Exposed?

its ``binary(1)``

**Comment by Tapac** [comment]
@NathanPB, first of all in exposed `binary` column type will be represented as `VARBINARY` not `BIT`, so your case with `BIT(1) is similar to `boolean` column type.
But there is indeed a problem with representing and comparing byteArray values which will be fixed in the next release (0.13.1)
